src/main/java/york/pharmacy/orders/Order.java

```diff
@@ -0,0 +1,46 @@
+package york.pharmacy.orders;
+
+import jakarta.persistence.*;
+import jakarta.validation.constraints.NotNull;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.CreatedDate;
+import org.springframework.data.annotation.LastModifiedDate;
+import org.springframework.data.jpa.domain.support.AuditingEntityListener;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@EntityListeners(AuditingEntityListener.class)
+public class Order {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+
+    // Blocked - Awaiting Medicine ID Foreign Key relation to the Medication Inventory Table
+    @NotNull
+    private String medicineId;
+
+    @NotNull
+    private int quantity;
+
+    @NotNull
+    private LocalDate deliveryDate;
+
+    @NotNull
+    @Enumerated(EnumType.STRING)
+    private OrderStatus status;
+
+    @CreatedDate
+    @Column(updatable = false)
+    private LocalDateTime createdAt;
```

Comment:
We should avoid `LocalDateTime` for auditing fields. The `LocalDateTime` doesn't have a timezone associated, so there's no way to tell the order of updates if they occurred from different timezones.